internal/provider/resource_time_offset.go

```diff
@@ -0,0 +1,495 @@
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var (
+	_ resource.Resource                = (*timeOffsetResource)(nil)
+	_ resource.ResourceWithImportState = (*timeOffsetResource)(nil)
+	_ resource.ResourceWithModifyPlan  = (*timeOffsetResource)(nil)
+)
+
+func NewTimeOffsetResource() resource.Resource {
+	return &timeOffsetResource{}
+}
+
+type timeOffsetResource struct{}
+
+func (t timeOffsetResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_offset"
+}
+
+func (t timeOffsetResource) GetSchema(ctx context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Description: "Manages an offset time resource, which keeps an UTC timestamp stored in the Terraform state that is" +
+			" offset from a locally sourced base timestamp. This prevents perpetual differences caused " +
+			"by using the [`timestamp()` function](https://www.terraform.io/docs/configuration/functions/timestamp.html).",
+		Attributes: map[string]tfsdk.Attribute{
+			"base_rfc3339": {
+				Description: "Base timestamp in " +
+					"[RFC3339](https://datatracker.ietf.org/doc/html/rfc3339#section-5.8) format " +
+					"(see [RFC3339 time string](https://tools.ietf.org/html/rfc3339#section-5.8) e.g., " +
+					"`YYYY-MM-DDTHH:MM:SSZ`). Defaults to the current time.",
+				Type:     types.StringType,
+				Optional: true,
+				Computed: true,
+			},
+			"day": {
+				Description: "Number day of offset timestamp.",
+				Type:        types.Int64Type,
+				Computed:    true,
+			},
+			"hour": {
+				Description: "Number hour of offset timestamp.",
+				Type:        types.Int64Type,
+				Computed:    true,
+			},
+			"triggers": {
+				Description: "Arbitrary map of values that, when changed, will trigger a new base timestamp value " +
+					"to be saved. See [the main provider documentation](../index.md) for more information.",
+				Type: types.MapType{
+					ElemType: types.StringType,
+				},
+				Optional: true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					resource.RequiresReplace(),
+				},
+			},
+			"minute": {
+				Description: "Number minute of offset timestamp.",
+				Type:        types.Int64Type,
+				Computed:    true,
+			},
+			"month": {
+				Description: "Number month of offset timestamp.",
+				Type:        types.Int64Type,
+				Computed:    true,
+			},
+			"offset_days": {
+				Description: "Number of days to offset the base timestamp. At least one of the 'offset_' arguments must be configured.",
+				Type:        types.Int64Type,
+				Optional:    true,
+				Validators: []tfsdk.AttributeValidator{
+					schemavalidator.AtLeastOneOf(path.MatchRoot("offset_years"),
+						path.MatchRoot("offset_months"),
+						path.MatchRoot("offset_hours"),
+						path.MatchRoot("offset_minutes"),
+						path.MatchRoot("offset_seconds")),
+				},
```

Comment:
Nit: If we would like to remove declaring this across all the offset attributes, it can be done by adding a declarative configuration validation method, e.g. [`resourcevalidator.AtLeastOneOf`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-validators@v0.5.0/resourcevalidator#AtLeastOneOf)

```go
import (
  "github.com/hashicorp/terraform-plugin-framework-validators/resourcevalidator"
)

var (
  _ resource.ResourceWithConfigValidators  = (*timeOffsetResource)(nil)
)

func (r timeOffsetResource) ConfigValidators(_ context.Context) []resource.ConfigValidator {
  return []resource.ConfigValidator{
    resourcevalidator.AtLeastOneOf(
      path.MatchRoot("offset_hours"),
      path.MatchRoot("offset_minutes"),
      path.MatchRoot("offset_months"),
      path.MatchRoot("offset_seconds"),
      path.MatchRoot("offset_years"),
    ),
  }
}
```